packages/components/addon/components/hds/button/index.hbs

```diff
@@ -1,10 +1,4 @@
-<button
-  class={{this.classNames}}
-  ...attributes
-  aria-label={{if this.isIconOnly this.text null}}
-  type={{this.type}}
-  disabled={{if this.isDisabled "disabled" null}}
->
+<button class={{this.classNames}} ...attributes aria-label={{if this.isIconOnly this.text null}} type={{this.type}}>
```

Comment:
🤔 💭 @didoo what's your instinct on making this a little bit more flexible in terms of a11y, by allowing consumers of this component to override the `aria-label` explicitly via splattributes? _(BTW I think it makes perfect sense to constrain the supported values for `type`, my question's _only_ about `arial-label`.)_

```suggestion
<button class={{this.classNames}} aria-label={{if this.isIconOnly this.text null}} ...attributes type={{this.type}}>
```

As-is RN, I believe the ordering of this attribute VS splattributes would always override a `arial-label="My special case a11y label ..."`, this might be unexpected now that folks can use `disabled`.

Reply:
@jesdavpet this specific code was created by @MelSumner to force the consumers to add a label when the button is icon-only, but I see your point.

@MelSumner thoughts on this?

Reply:
> _"... to force the consumers to add a label when the button is icon-only ..."_

Makes sense, but I don't think they're mutually exclusive. 🙂 The proposed change in ordering would _not_ stop the override from working in the case of an icon-only button ... it would just allow the component consumer to override the `aria-label` if/when necessary.

Reply:
@jesdavpet I'm not sure I understand your goal, so let's break it down: 

1. right now, the developer has to set the value of `@text` which, if it's an icon-only button, becomes the value of `aria-label`. 

In what cases would you need to override what's already provided? Seems like you're overriding...yourself? Am I missing something?

Reply:
Also, let's break this out into a different issue so we don't block this one for an unrelated reason.

Reply:
@MelSumner the scenario I've raised is that of a component consumer who wants to set a specific value for `aria-label` for a an invocation that _isn't_ "icon-only".

Reply:
> _"Also, let's break this out into a different issue..."_

This is the _only_ topic of the issue I started, it's non-blocking.

Reply:
@MelSumner what @jesdavpet is suggesting is to allow users to do something like this
```
<Hds::Button @icon="play" @text="this is the visible text" aria-label="this is the additional aria-label that for some reason they may want to add" ... />
```
with the current implementation, this is not possible. I can imagine @jesdavpet position: having to write a codemod that takes care of all possible use cases, and considering the possibility that this may be some strange exception in some parts of the codebase, having it solved before doing the migration would be a plus. if you think it can't be done/decided in the context of this PR, I can open a new one.

Reply:
@didoo @jesdavpet ahh yes okay. So we don't allow that because it's a [failure due to the accessible name not containing the visible label text](https://www.w3.org/WAI/WCAG21/Techniques/failures/F96), a violation of [2.5.3- Label in Name](https://www.w3.org/WAI/WCAG21/Understanding/label-in-name).